cmd/keeper/keeper.go

```diff
@@ -842,6 +842,80 @@ func (p *PostgresKeeper) updateReplSlots(dbLocalState *DBLocalState, followersUI
 	return nil
 }
 
+func (p *PostgresKeeper) recreateAddonReplicationSlots(addonSlotNames []string) error {
+	var existingReplicationSlots []string
+	existingReplicationSlots, err := p.pgm.GetReplicationSlots()
+
+	if err != nil {
+		log.Errorw("failed to get replication slots", zap.Error(err))
+		return err
+	}
+
+	// detect not stolon replication slots
+	notStolonSlots := []string{}
+	for _, currReplicationSlotName := range existingReplicationSlots {
+		if !common.IsStolonName(currReplicationSlotName) {
+			notStolonSlots = append(notStolonSlots, currReplicationSlotName)
+		}
+	}
+
+	// detect slot that should be dropped
+	slotsToDrop := []string{}
+
+	for _, removeCandidateSlot := range notStolonSlots {
+		slotShouldSurvive := false
+		for _, replSlotName := range addonSlotNames {
+			if replSlotName == removeCandidateSlot {
+				slotShouldSurvive = true
+			}
+		}
+
+		if !slotShouldSurvive { // slot not detected as additional, so schedule the removal
+			slotsToDrop = append(slotsToDrop, removeCandidateSlot)
+		}
+	}
+
+	// detect slots that should be created
+	slotsToCreate := []string{}
+
+	for _, addReplicationSlotName := range addonSlotNames {
+		slotExists := false
+
+		for _, currReplicationSlotName := range notStolonSlots {
+			if addReplicationSlotName == currReplicationSlotName {
+				slotExists = true
+				break
+			}
+		}
+		if !slotExists {
+			slotsToCreate = append(slotsToCreate, strings.TrimSpace(addReplicationSlotName))
+		}
+	}
+
+	// drop unnecessary slots
+	if len(slotsToDrop) > 0 {
+		for _, dropSlotName := range slotsToDrop {
+			log.Infow("Drop slot", "dropSlotName", dropSlotName)
+			if err = p.pgm.DropReplicationSlot(dropSlotName); err != nil {
+				log.Errorw("Failed to drop replication slot", "slotName", dropSlotName, zap.Error(err))
+				return err
```

Comment:
ignore dropping errors like done in `updateReplSlot` since they can be in use and this will fail.